src/Microsoft.DotNet.Build.Tasks/PackageFiles/init-tools.cmd

```diff
@@ -14,7 +14,11 @@ set BUILDTOOLS_PACKAGE_DIR=%~dp0
 set MICROBUILD_VERSION=0.2.0
 set PORTABLETARGETS_VERSION=0.1.1-dev
 set ROSLYNCOMPILERS_VERSION=2.0.0-rc
-set MSBUILD_CONTENT_JSON={"dependencies": { "MicroBuild.Core": "%MICROBUILD_VERSION%", "Microsoft.Portable.Targets": "%PORTABLETARGETS_VERSION%", "Microsoft.Net.Compilers": "%ROSLYNCOMPILERS_VERSION%"},"frameworks": {"netcoreapp1.0": {},"net46": {}}}
+if [%BUILDTOOLS_USE_CSPROJ%]==[] (
+  set MSBUILD_PROJECT_CONTENT={"dependencies": { "MicroBuild.Core": "%MICROBUILD_VERSION%", "Microsoft.Portable.Targets": "%PORTABLETARGETS_VERSION%", "Microsoft.Net.Compilers": "%ROSLYNCOMPILERS_VERSION%"},"frameworks": {"netcoreapp1.0": {},"net46": {}}}
+) ELSE (
+  set MSBUILD_PROJECT_CONTENT=^^^<Project Sdk="Microsoft.NET.Sdk"^^^>^^^<PropertyGroup^^^>^^^<TargetFrameworks^^^>netcoreapp1.0;net46^^^</TargetFrameworks^^^>^^^<DisableImplicitFrameworkReferences^^^>true^^^</DisableImplicitFrameworkReferences^^^>^^^</PropertyGroup^^^>^^^<ItemGroup^^^>^^^<PackageReference Include="MicroBuild.Core" Version="%MICROBUILD_VERSION%" /^^^>^^^<PackageReference Include="Microsoft.Portable.Targets" Version="%PORTABLETARGETS_VERSION%" /^^^>^^^<PackageReference Include="Microsoft.Net.Compilers" Version="%ROSLYNCOMPILERS_VERSION%" /^^^>^^^</ItemGroup^^^>^^^</Project^^^>
```

Comment:
For the msbuild project I wouldn't actually do it the same way. Instead I would check it in as a file and just pass the versions as properties when you do the restore. The only reason we ever generated it here is because we need the version to be in-sync but msbuild has support for doing this by passing in the property.